packages/apps/transfer/src/services/cross-chain-transfer-finish/finish-xchain-transfer.ts

```diff
@@ -0,0 +1,82 @@
+import { ChainwebNetworkId } from '@kadena/chainweb-node-client';
+import { ContCommand, getContCommand } from '@kadena/client';
+import { ChainId } from '@kadena/types';
+
+import { generateApiHost } from '../utils/utils';
+
+interface ITransactionData {
+  sender: { chain: ChainId; account: string };
+  receiver: { chain: ChainId; account: string };
+  amount: number;
+  receiverGuard: {
+    pred: string;
+    keys: [string];
+  };
+}
+export interface ITransferDataResult {
+  tx?: ITransactionData | undefined;
+  error?: string;
+}
+
+export interface TransferResult {
+  requestKey?: string;
+  status?: string;
+}
+
+const gasLimit: number = 850;
+const gasPrice: number = 0.00000001;
+
+export async function finishXChainTransfer(
+  requestKey: string,
+  step: number,
+  pactID: string,
+  rollback: boolean,
+  server: string,
+  network: ChainwebNetworkId,
+  targetChain: ChainId,
+  gasPayer: string,
+): Promise<ContCommand | { error: string }> {
+  const host = generateApiHost(server, network, targetChain);
+  const hostSPV = `${generateApiHost(server, network, '1')}/spv`;
+
+  try {
+    const contCommand = await getContCommand(
+      requestKey,
+      targetChain,
+      hostSPV,
+      step + 1,
+      rollback,
+    );
+
+    contCommand.setMeta(
+      {
+        chainId: targetChain,
+        sender: gasPayer,
+        gasLimit,
+        gasPrice,
+      },
+      network,
+    );
+
+    // remove double quotes from stringified proof
+    contCommand.proof = contCommand.proof.slice(1, -1);
```

Comment:
I don't know the proof format but wouldn't it better to do `.replaceAll('"', '')` just in case the api response ever changes to not include quotes?

Reply:
I think this should be changed on the client itself. I will make the change. I also ran into the same issue and I don't think its sensible to have to slice or replace this manually.